Package is broken with the latest commit

Hello @elazarl ,

I am trying to install the latest version of this package on go-1.14 and go-1.14.1 and getting error:

```
$ go get -u github.com/elazarl/goproxy
go: downloading github.com/elazarl/goproxy v0.0.0-20200315184450-1f3cb6622dad
go: github.com/elazarl/goproxy upgrade => v0.0.0-20200315184450-1f3cb6622dad
go: downloading golang.org/x/crypto v0.0.0-20200128174031-69ecbb4d6d5d
go: downloading golang.org/x/net v0.0.0-20191126235420-ef20fe5d7933
go: downloading golang.org/x/text v0.3.3-0.20191031172631-4b67af870c6f
go: downloading golang.org/x/sys v0.0.0-20200201011859-915c9c3d4ccf
go: golang.org/x/text upgrade => v0.3.2
go: golang.org/x/sys upgrade => v0.0.0-20200327173247-9dae0f8f5775
go: golang.org/x/crypto upgrade => v0.0.0-20200323165209-0ec3e9974c59
go: golang.org/x/net upgrade => v0.0.0-20200324143707-d3edc9973b7e
go: downloading golang.org/x/crypto v0.0.0-20200323165209-0ec3e9974c59
go: downloading golang.org/x/net v0.0.0-20200324143707-d3edc9973b7e
go: downloading golang.org/x/sys v0.0.0-20200327173247-9dae0f8f5775
# github.com/elazarl/goproxy
../../../../pkg/mod/github.com/elazarl/goproxy@v0.0.0-20200315184450-1f3cb6622dad/ctx.go:46:2: not enough arguments to return
../../../../pkg/mod/github.com/elazarl/goproxy@v0.0.0-20200315184450-1f3cb6622dad/ctx.go:46:12: ctx.proxy undefined (type *ProxyCtx has no field or method proxy, but does have Proxy)
../../../../pkg/mod/github.com/elazarl/goproxy@v0.0.0-20200315184450-1f3cb6622dad/ctx.go:50:5: ctx.proxy undefined (type *ProxyCtx has no field or method proxy, but does have Proxy)
../../../../pkg/mod/github.com/elazarl/goproxy@v0.0.0-20200315184450-1f3cb6622dad/ctx.go:62:8: ctx.proxy undefined (type *ProxyCtx has no field or method proxy, but does have Proxy)
../../../../pkg/mod/github.com/elazarl/goproxy@v0.0.0-20200315184450-1f3cb6622dad/https.go:76:69: unknown field 'proxy' in struct literal of type ProxyCtx (but does have Proxy)
../../../../pkg/mod/github.com/elazarl/goproxy@v0.0.0-20200315184450-1f3cb6622dad/https.go:199:77: unknown field 'proxy' in struct literal of type ProxyCtx (but does have Proxy)
../../../../pkg/mod/github.com/elazarl/goproxy@v0.0.0-20200315184450-1f3cb6622dad/proxy.go:105:70: unknown field 'proxy' in struct literal of type ProxyCtx (but does have Proxy)


```

--- a/ctx.go
+++ b/ctx.go
@@ -43,11 +43,11 @@ func (ctx *ProxyCtx) RoundTrip(req *http.Request) (*http.Response, error) {
 	if ctx.RoundTripper != nil {
 		return ctx.RoundTripper.RoundTrip(req, ctx)
 	}
-	return ctx.proxy.Tr.RoundTrip(req)
+	return ctx.Proxy.Tr.RoundTrip(req)
 }
 
 func (ctx *ProxyCtx) printf(msg string, argv ...interface{}) {
-	ctx.proxy.Logger.Printf("[%03d] "+msg+"\n", append([]interface{}{ctx.Session & 0xFF}, argv...)...)
+	ctx.Proxy.Logger.Printf("[%03d] "+msg+"\n", append([]interface{}{ctx.Session & 0xFF}, argv...)...)
 }
 
 // Logf prints a message to the proxy's log. Should be used in a ProxyHttpServer's filter
@@ -59,7 +59,7 @@ func (ctx *ProxyCtx) printf(msg string, argv ...interface{}) {
 //		return r, nil
 //	})
 func (ctx *ProxyCtx) Logf(msg string, argv ...interface{}) {
-	if ctx.proxy.Verbose {
+	if ctx.Proxy.Verbose {
 		ctx.printf("INFO: "+msg, argv...)
 	}
 }

--- a/https.go
+++ b/https.go
@@ -73,7 +73,7 @@ type halfClosable interface {
 var _ halfClosable = (*net.TCPConn)(nil)
 
 func (proxy *ProxyHttpServer) handleHttps(w http.ResponseWriter, r *http.Request) {
-	ctx := &ProxyCtx{Req: r, Session: atomic.AddInt64(&proxy.sess, 1), proxy: proxy, certStore: proxy.CertStore}
+	ctx := &ProxyCtx{Req: r, Session: atomic.AddInt64(&proxy.sess, 1), Proxy: proxy, certStore: proxy.CertStore}
 
 	hij, ok := w.(http.Hijacker)
 	if !ok {
@@ -196,7 +196,7 @@ func (proxy *ProxyHttpServer) handleHttps(w http.ResponseWriter, r *http.Request
 			clientTlsReader := bufio.NewReader(rawClientTls)
 			for !isEof(clientTlsReader) {
 				req, err := http.ReadRequest(clientTlsReader)
-				var ctx = &ProxyCtx{Req: req, Session: atomic.AddInt64(&proxy.sess, 1), proxy: proxy, UserData: ctx.UserData}
+				var ctx = &ProxyCtx{Req: req, Session: atomic.AddInt64(&proxy.sess, 1), Proxy: proxy, UserData: ctx.UserData}
 				if err != nil && err != io.EOF {
 					return
 				}

--- a/proxy.go
+++ b/proxy.go
@@ -102,7 +102,7 @@ func (proxy *ProxyHttpServer) ServeHTTP(w http.ResponseWriter, r *http.Request) 
 	if r.Method == "CONNECT" {
 		proxy.handleHttps(w, r)
 	} else {
-		ctx := &ProxyCtx{Req: r, Session: atomic.AddInt64(&proxy.sess, 1), proxy: proxy}
+		ctx := &ProxyCtx{Req: r, Session: atomic.AddInt64(&proxy.sess, 1), Proxy: proxy}
 
 		var err error
 		ctx.Logf("Got request %v %v %v %v", r.URL.Path, r.Host, r.Method, r.URL.String())

